### ISSUE TITLE
FUEL: Error creating vocab (no-vocab) when running fuel-scaffold-vocab

### ISSUE THREAD

**Issue by amherag** [factor/factor#1996]
When in Emacs, I run `fuel-scaffold-vocab`, I answer the questions, fill in the summary, tags, etc. and then I get this error in the mini-buffer:

```
Error creating vocab (no-vocab)
```

A directory is indeed created in the `work` resource, but I only get a .factor file; no docs, summary or tags, tests files, only the "main" source code file.

**Comment by catb0t** [comment]
What version / git tag of Factor? `HEAD`? Or older?

**Comment by amherag** [comment]
HEAD.

**Comment by amherag** [comment]
Actually, I ran:

```
USE: tools.scaffold
"palindrome" scaffold-work
```

in a new listener and I get the error: `Vocabulary does not exist`.

If I run `"palindrome" scaffold-work` again I get the error `vocab-must-not-exist`.

**Comment by mrjbq7** [comment]
Yes, it seems a (simple?) race condition in ``scaffold-vocab``, but I haven't had time to figure it entirely out yet.

```factor
: scaffold-vocab ( vocab-root string -- )
    check-vocab-exists {
        [ scaffold-directory ]
        [ scaffold-main ]
        [ nip require ]
        [ nip scaffold-authors ]
    } 2cleave ;
```

And in fact, if you run those one at a time, it works fine, but in a single word, seems to fail when it goes to create the authors file, I think.  Basically, the ``require`` didn't seem to configure the vocab for the subsequence call to ``scaffold-authors``. 

I've been traveling a little bit lately, and have been meaning to fix this unless @erg gets to it before me.

**Comment by catb0t** [comment]
It's not a problem with `scaffold-vocab` itself because changing it to use locals and do exactly one thing at a time:

```factor
:: scaffold-vocab ( vocab-root name -- )
    name check-vocab-exists drop
    vocab-root name scaffold-directory
    vocab-root name scaffold-main
    name scaffold-authors
    name require
    ;
``` 

gives the same "Vocabulary does not exist".

**Comment by jonenst** [comment]
2cleave already does one thing at a time. Your version with locals should compile to mostly the same code as the one with 2cleave (you should be able to see that with compiler.tree.debugger => optimized. or tools.disassembler => disassemble)

**Comment by mrjbq7** [comment]
I have a fix for this almost ready.

Sorry about the bug!

The ``check-vocab-exists`` word pollutes the ``root-cache`` (it causes it
to cache an ``f`` for no root found) which causes a later require to fail
with ``no-vocab``.

**Comment by mrjbq7** [comment]
Broken in 68eb6833872571a8e931cf2246838e5e0e619052
